types/providers.d.ts

```diff
@@ -33,6 +33,7 @@ export interface OAuthConfig<P extends Record<string, unknown> = Profile>
   requestTokenUrl?: string
   authorizationUrl: string
   profileUrl: string
+  authHeaders?(info: { accessToken: string }): Record<string, string>,
```

Comment:
You get any tokens returned from the provider in the second param of the `profile` callback:
```ts
async profile(profile, tokens) {
  const data = await getDataFromApi(tokens.access_token)
  return {
    id: data.sub,
    name: data.userName,
    image: data.picture
  }
}
// ...
async getDataFromApi(token) {
  return fetch('path/to/api', {
    headers: {
      'X-Auth-Token': token
    }
  })
} 
```
Users can use that to make API calls to the provider. This is [documented in our options](https://next-auth.js.org/configuration/providers#options), although maybe we should highlight it better 🤔 as it's a common use case.

Reply:
This is used in https://github.com/djyde/next-auth/blob/69f1559ced9bc130238fd09c85c3a01578c9e6eb/src/server/lib/oauth/client.js#L222